docs/src/main/asciidoc/datasource.adoc

```diff
@@ -99,161 +165,209 @@ To install the PostgreSQL driver dependency for instance, just run the following
+you won't need to configure the driver.
+====
 
-PostgreSQL only runs as a server, as do the rest of the databases below.
-As such, you must specify connection details, or use the defaults.
+Note that you can still define a specific driver if you need to (for instance for using the OpenTracing driver)
+or if you want to use a database for which we don't have a built-in JDBC driver extension.
 
-`jdbc:postgresql:[//][host][:port][/database][?key=value...]`
+Defining a specific JDBC driver is as simple as:
 
-Example:: `jdbc:postgresql://localhost/test`
+[source, properties]
+----
+quarkus.datasource.jdbc.driver=io.opentracing.contrib.jdbc.TracingDriver
+----
 
-Defaults for the different parts are as follows:
+==== More configuration
 
-`host`:: localhost
-`port`:: 5432
-`database`:: same name as the username
+You can configure a lot more things, for instance the size of the connection pool.
 
-The https://jdbc.postgresql.org/documentation/head/connect.html[official documentation] go into more detail and list optional parameters as well.
+Please refer to the <<jdbc-configuration,JDBC configuration reference>> for all the details about the JDBC configuration knobs.
 
-==== MariaDB
+==== Consuming the datasource
 
-`jdbc:mariadb:[replication:|failover:|sequential:|aurora:]//<hostDescription>[,<hostDescription>...]/[database][?<key1>=<value1>[&<key2>=<value2>]]`
- hostDescription:: `<host>[:<portnumber>]  or address=(host=<host>)[(port=<portnumber>)][(type=(master|slave))]`
+If you are using Hibernate ORM, the datasource will be consumed automatically.
 
-Example:: `jdbc:mariadb://localhost:3306/test`
+If you want control over the datasource, Quarkus uses CDI so you can inject the datasource in any managed bean:
 
-You can find more information about this feature and others detailed in the https://mariadb.com/kb/en/library/about-mariadb-connector-j/[official documentation].
+[source,java]
+----
+@Inject
+AgroalDataSource defaultDataSource;
+----
 
-==== MySQL
+In the above example, the type is `AgroalDataSource` which is a subtype of `javax.sql.DataSource`.
+Because of this, you can also use `javax.sql.DataSource` as the injected type.
 
-`jdbc:mysql:[replication:|failover:|sequential:|aurora:]//<hostDescription>[,<hostDescription>...]/[database][?<key1>=<value1>[&<key2>=<value2>]]`
- hostDescription:: `<host>[:<portnumber>]  or address=(host=<host>)[(port=<portnumber>)][(type=(master|slave))]`
+=== Reactive datasource
 
-Example:: `jdbc:mysql://localhost:3306/test`
+If you prefer using a reactive datasource, Quarkus offers MySQL/MariaDB and PostgreSQL reactive clients.
 
-You can find more information about this feature and others detailed in the https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference.html[official documentation].
+==== Install the Maven dependencies
 
-==== Microsoft SQL Server
+Depending on if you wish to use MySQL/MariaDB or PostgreSQL, you need to add either the `quarkus-reactive-pg-client`
```

Comment:
```suggestion
Depending on whether wish to use MySQL/MariaDB or PostgreSQL, you need to add either the `quarkus-reactive-pg-client`
```

Reply:
@geoand should it be `Depending on whether you wish ..` ?

Reply:
Yup, updated.

Don't be surprised when you see my suggestions miss words. That happens to me very often...